src/interfaces/mining.h

```diff
@@ -5,26 +5,44 @@
 #ifndef BITCOIN_INTERFACES_MINING_H
 #define BITCOIN_INTERFACES_MINING_H
 
+#include <consensus/amount.h>
+#include <primitives/transaction.h>
 #include <node/types.h>
 #include <uint256.h>
 
 #include <memory>
 #include <optional>
 
 namespace node {
-struct CBlockTemplate;
 struct NodeContext;
 } // namespace node
 
 class BlockValidationState;
 class CBlock;
+class CBlockHeader;
 class CScript;
 
 namespace interfaces {
 
+//! Block template interface
+class BlockTemplate
```

Comment:
In commit "Have createNewBlock return BlockTemplate interface" (977ab92f8040cd139fc4b969baeec39ef58c00af)

Want to note that at some point, depending on how this interface is used, it might be a good idea to consolidate individual accessor methods into more general ones that return structs or tuples to return multiple values at the same time.

For example, if getTxFees() and getTxSigops() are typically called one after another over IPC it could make sense to combine them into a `getTxInfo()` function to save a round trip.

Reply:
I'm not sure if we end using `getTxFees()` and `getTxSigops()` in stratum v2, or only for the existing stratum v1 RPC calls. Once sv2 code is a bit more mature it's indeed worth going over the interface once more.